test/Mono.Linker.Tests.Cases/DataFlow/AssemblyQualifiedNameDataflow.cs

```diff
@@ -50,6 +51,14 @@ static void TestConstructors ()
 			RequireNothing (type);
 		}
 
+		[ExpectedWarning ("IL2104",
+			"Type 'System.Invalid.TypeName' was not found in the caller assembly nor in the base library. " +
+			"Type name strings used for dynamically accessing a type should be assembly qualified.")]
+		static void TestUnqualifiedTypeNameWarns ()
+		{
+			RequirePublicConstructors ("System.Invalid.TypeName");
+		}
+
```

Comment:
Please also add a test for this happening in the `TypeConverter` attribute - as that is a special code path in the mark step (doesn't go through data flow).

I filed https://github.com/mono/linker/issues/1942 to remove this as a special case, but for now we should add a test for this.